src/nvim/ex_docmd.c

```diff
@@ -5640,6 +5641,20 @@ static void ex_syncbind(exarg_T *eap)
   }
 }
 
+void do_read_cmd(exarg_T *eap)
+{
+  char *cmd = eap->arg + 1;
+  StringBuilder put_cmd = KV_INITIAL_VALUE;
+  kv_printf(put_cmd, "try | ");
+  kv_printf(put_cmd, "%dput=execute('%s') | ", eap->line2, cmd);
```

Comment:
will need to escape single-quotes, possibly other stuff. actually, all that could be avoided, including support for double quotes ("), by storing `cmd` in a temporary `g:_blah` var, then doing:

```suggestion
  kv_printf(put_cmd, "%dput=execute(g:_blah) | ", eap->line2, cmd);
```

Reply:
OK, I am now using an api call to put {cmd} directly into a global variable. I haven't updated the unit tests yet though as I think I am going to have to take some time to understand how things are being parsed before being to `ex_read`. For example, the big issue with double quotes (as I understand it right now) is that these cause the text to be marked as a comment and chopped off before `eap` is passed to `ex_read`.

Reply:
yeah it's possible that double quotes can't easily be supported because the cmdline is parsed before control reaches the code you have added here. can forget about that for now.